### ISSUE TITLE
Font Suggestion: Source Han Sans

### ISSUE THREAD

**Issue by TriforceOfKirby** [alfredxing/brick#108]
https://github.com/adobe-fonts/source-han-sans

**Comment by alfredxing** [comment]
Thanks! There's a huge backlog of fonts already; I don't think I'll be able to start working on adding fonts until my exams are done (start of May or start of June).

**Comment by TriforceOfKirby** [comment]
I could help add fonts if you want via pull request.

**Comment by alfredxing** [comment]
@TriforceOfKirby It is a bit tricky, but if you want to tackle it, that would be great!

**Comment by TriforceOfKirby** [comment]
@alfredxing Ok so I opened the font in FontForge, then went to `File>Generate Fonts...` selected `Web Open Font` and clicked `Generate`.

It gives me a warning message:

> You are attempting to save a CID font in a non-CID format.
> This is ok, but it means that only the current sub-font will be saved. Is that what you want?

If I click `Yes`, it also gives me another warning message about the em size being 1000 and that it should normally be a power of 2.

Clicking `Yes` again results in a woff format font that is 5kB(5,088 bytes) file, the original font in otf format is 16.4MB(16,421,764 bytes).

I'm not sure if this is right, I don't think the font file size should shrink _that_ much.

**Comment by alfredxing** [comment]
@TriforceOfKirby Yeah, I think it's a problem with this being an Asian font, which might have different data than regular fonts...

I haven't worked with CID fonts before, but maybe try flattening them? [This page](http://fontforge.github.io/cidmenu.html) provides docs on manipulating CID fonts in FontForge.

**Comment by TriforceOfKirby** [comment]
@alfredxing So flattening seems to work, but I do get a couple warning messages:

>    Internal Error:
>    Atempt to output -63677 into a 16-bit field. It will be truncated and the file may not be useful.

I still get the font file, but I'm not sure if it was created correctly. I made a test web page with the font embedded on [Dropbox](https://db.tt/shTyTxjy).

**Comment by alfredxing** [comment]
That should be okay. It'll just be one character probably, and there's not much we can do other than that. Did the em-size error go away?

Thanks for doing this, by the way. It really helps out!

**Comment by TriforceOfKirby** [comment]
@alfredxing No, the em-size warning still shows. Although, [this page](http://fontforge.github.io/overview.html#em-unit) on FontForge's website says that 1000 is a perfectly fine em size.

You're welcome! I'll help out with a few more fonts too.

**Comment by alfredxing** [comment]
Yep, 1000 is fine. I think (though I'm not sure) that 1000 is used for PostScript outlines, and a power of two (like 1024 or 2048) is usually used for TrueType.

**Comment by TriforceOfKirby** [comment]
@alfredxing So I found a command-line tool by Mozilla for converting sfnt fonts(otf, ttf, etc.) to woff format [here](https://people.mozilla.org/~jkew/woff/).

With that I ran this batch command:
    FOR %A IN (*.otf) DO sfnt2woff %A
And it converted all the files without any errors, so would this method of converting the fonts be ok?
